File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mini = prices[0];
        int n =prices.size();
        int maxi =0;
        for(int i=0;i<n;i++){
            int cost = prices[i]-mini;
            maxi = max(maxi,cost);
            mini = min(mini,prices[i]);
        }return maxi;

    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mini = prices[0]
        maxpro = 0
        for i in range(1,len(prices)):
            cost =prices[i] - mini
            maxpro = max(maxpro,cost)
            mini = min(mini,prices[i])
        return maxpro